qiskit_ibm_provider/transpiler/passes/scheduling/block_base_padder.py

```diff
@@ -420,7 +437,7 @@ def _visit_control_flow_op(self, node: DAGNode) -> None:
         self._add_block_terminating_barrier(block_idx, t0, node)
 
         # Only pad non-fast path nodes
-        fast_path_node = node in self._fast_path_nodes
+        fast_path_node = node in self._fast_path_nodes or not self._schedule_idle_qubits
```

Comment:
only nitpicking here, perhaps rename `fast_path_node` to `skip_padding` or something like that, as there are two different conditions for that to happen

Reply:
On a second thought, I'm not sure we want to disable padding within the control flow here. As a result of this (and related changes at L163-169, this circuit will miss padding (and potentially DD) in the conditional block:
```
creg = ClassicalRegister(1)
qreg = QuantumRegister(3)
circ = QuantumCircuit(qreg,creg)
circ.sx(qreg[0])
circ.x(qreg[1])
circ.sx(qreg[2])
circ.barrier()
circ.measure(0,creg[0])
with circ.if_test((creg[0], 0)):
    circ.x(qreg[1])
```
![missing_padding](https://github.com/Qiskit/qiskit-ibm-provider/assets/10402430/bf3ac00c-864d-4f2e-a92d-a449b74c8cf2)

Whereas the expected behavior should be (reverting those changes):
![correct_padding](https://github.com/Qiskit/qiskit-ibm-provider/assets/10402430/8e80c935-5df8-4187-96ec-2f633371b7f1)